rfc/helm-import.md

```diff
@@ -0,0 +1,515 @@
+  const minor = helmChartDetails[4];
+  const patch = helmChartDetails[5];
+
+  const chartVersion = `${major}.${minor}.${patch}`;
+
+  return [chartUrl, chartName, chartVersion];
+}
+```
+
+##### Getting `values.schema.json` file
+
+When the user would run the import command, we would need to identify if a `values.schema.json` is present in the helm chart being mentioned. To do so, we can create a temporary directory and pull in and extract the chart in that directory. 
+
+```typescript
+/**
+ * Pulls the helm chart in a temporary directory
+ * @param chartUrl Chart url
+ * @param chartName Chart name
+ * @param chartVersion Chart version
+ * @returns Temporary directory path
+ */
+function pullHelmRepo(chartUrl: string, chartName: string, chartVersion: string): string {
+  const workdir = fs.mkdtempSync(path.join(os.tmpdir(), 'cdk8s-helm-'));
+
+  const args = new Array<string>();
+  args.push('pull');
+  args.push(chartName);
+  args.push('--repo', chartUrl);
+  args.push('--version', chartVersion);
+  args.push('--untar');
+  args.push('--untardir', workdir);
+
+  const command = 'helm';
+
+  const helm = spawnSync(command, args, {
+    maxBuffer: MAX_HELM_BUFFER,
+  });
+
+  if (helm.error) {
+    const err = helm.error.message;
+    if (err.includes('ENOENT')) {
+      throw new Error(`unable to execute '${command}' to pull the Helm chart. Is it installed on your system?`);
+    }
+
+    throw new Error(`error while pulling a helm chart: ${err}`);
+  }
+
+  if (helm.status !== 0) {
+    throw new Error(helm.stderr.toString());
+  }
+
+  return workdir;
+}
+```
+
+After extraction, we would look for the `values.schema.json` in the root of the helm chart.
+
+```typescript
+  if (fs.existsSync(this.tmpDir)) {
+    this.chartSchemaPath = path.join(this.tmpDir, this.chartName, this.schemaFileName);
+  } else {
+    this.chartSchemaPath = undefined;
+  }
+```
+
+> **Note:**
+>
+> After the typescript code generation is successful, we would cleanup the temporary directory created which had the helm chart.
+>   ```typescript
+>    /**
+>     * Cleanup temp directory created
+>     * @param tmpDir Temporary directory path
+>     */
+>    function cleanup(tmpDir: string) {
+>       fs.rmSync(tmpDir, { recursive: true });
+>    }
+>   ```
+
+#### Code Generation
+
+ImportBase class has abstract methods [moduleNames](https://github.com/cdk8s-team/cdk8s-cli/blob/2.x/src/import/base.ts#L40) and [generateTypeScript](https://github.com/cdk8s-team/cdk8s-cli/blob/2.x/src/import/base.ts#L42). As part of the new class `ImportHelm` we also need to define these. Because, when the [import](https://github.com/cdk8s-team/cdk8s-cli/blob/2.x/src/import/base.ts#L44) function is [invoked]((https://github.com/cdk8s-team/cdk8s-cli/blob/2.x/src/import/dispatch.ts#L17)), it invokes the [generateTypeScript](https://github.com/cdk8s-team/cdk8s-cli/blob/2.x/src/import/base.ts#L80-L82) function. 
+
+```typescript
+  public get moduleNames() {
```

Comment:
Can you explain what this method is used for and why you decided you return the chart name? is this the name of the generated `.ts` file?

Reply:
Yes, I used it here since this would be used for name of the generated file. Although, now I see that `module` does not sound correct for this.

Reply:
Link: https://github.com/cdk8s-team/cdk8s-cli/blob/2.x/src/import/base.ts#L40